From 25216311b6db92ae01473f54ad5564cc11507e97 Mon Sep 17 00:00:00 2001
From: Nicola Fontana <ntd@entidi.it>
Date: Mon, 23 Mar 2015 19:02:51 +0100
Subject: [PATCH 07/10] Allow overriding of os.name

os.name returns the OS running python, but in cross-compilation this
approach is flawed by design because the target OS can be different.

Allow overriding of such value with GI_OS_NAME environment variable.
---
 giscanner/dumper.py      | 2 +-
 giscanner/shlibs.py      | 4 ++--
 giscanner/transformer.py | 3 ++-
 giscanner/utils.py       | 5 ++++-
 4 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/giscanner/dumper.py b/giscanner/dumper.py
index 564be52..858d472 100644
--- a/giscanner/dumper.py
+++ b/giscanner/dumper.py
@@ -258,7 +258,7 @@ class DumpCompiler(object):
         else:
             args.extend(['-o', output])
         if libtool:
-            if os.name == 'nt':
+            if utils.osname() == 'nt':
                 args.append('-Wl,--export-all-symbols')
             else:
                 args.append('-export-dynamic')
diff --git a/giscanner/shlibs.py b/giscanner/shlibs.py
index a8e0bb9..df00980 100644
--- a/giscanner/shlibs.py
+++ b/giscanner/shlibs.py
@@ -24,7 +24,7 @@ import platform
 import re
 import subprocess
 
-from .utils import get_libtool_command, extract_libtool_shlib, which
+from .utils import get_libtool_command, extract_libtool_shlib, which, osname
 from .ccompiler import CCompiler
 
 
@@ -89,7 +89,7 @@ def _resolve_non_libtool(options, binary, libraries):
         else:
             binary.args[0] = old_argdir
 
-    if os.name == 'nt':
+    if osname() == 'nt':
         cc = CCompiler()
         shlibs = cc.resolve_windows_libs(libraries, options)
 
diff --git a/giscanner/transformer.py b/giscanner/transformer.py
index 8c5e908..c76ae9a 100644
--- a/giscanner/transformer.py
+++ b/giscanner/transformer.py
@@ -34,6 +34,7 @@ from .sourcescanner import (
     CSYMBOL_TYPE_ENUM, CSYMBOL_TYPE_UNION, CSYMBOL_TYPE_OBJECT,
     CSYMBOL_TYPE_MEMBER, CSYMBOL_TYPE_ELLIPSIS, CSYMBOL_TYPE_CONST,
     TYPE_QUALIFIER_CONST, TYPE_QUALIFIER_VOLATILE)
+from .utils import osname
 
 
 class TransformerException(Exception):
@@ -43,7 +44,7 @@ class TransformerException(Exception):
 _xdg_data_dirs = [x for x in os.environ.get('XDG_DATA_DIRS', '').split(os.pathsep)]
 _xdg_data_dirs.append(DATADIR)
 
-if os.name != 'nt':
+if osname() != 'nt':
     _xdg_data_dirs.append('/usr/share')
 
 
diff --git a/giscanner/utils.py b/giscanner/utils.py
index 0c6662f..ca3d45c 100644
--- a/giscanner/utils.py
+++ b/giscanner/utils.py
@@ -184,8 +184,11 @@ def cflag_real_include_path(cflag):
     return "-I" + os.path.realpath(cflag[2:])
 
 
+def osname():
+    return os.environ.get("GI_OS_NAME", os.name)
+
 def exeext():
-    if os.name == "nt":
+    if osname() == "nt":
         exeext = ".exe"
     else:
         exeext = ""
-- 
2.3.3

